crates/cli/src/main.rs

```diff
@@ -467,35 +468,68 @@ impl Commands {
-                                        std::process::exit(2);
+                        Some(job::Trigger::Options {
+                            push,
+                            pull_request,
+                            paths,
+                        }) => {
+                            match &*git_event {
+                                "pull_request" => {
+                                    if let Some(TriggerOn::Branches { branches }) = &pull_request {
+                                        if !branches.contains(&base_ref) {
+                                            info!(
+                                            "Skipping pipeline because branch {} is not in {}: {:?}",
+                                            base_ref.bold(),
+                                            "pull_request".bold(),
+                                            pull_request
+                                        );
+                                            std::process::exit(2);
+                                        }
                                     }
                                 }
-                            }
-                            "push" => {
-                                if let Some(TriggerOn::Branches { branches }) = &push {
-                                    if !branches.contains(&base_ref) {
-                                        info!(
+                                "push" => {
+                                    if let Some(TriggerOn::Branches { branches }) = &push {
+                                        if !branches.contains(&base_ref) {
+                                            info!(
                                         "Skipping pipeline because branch {} is not in {}: {:?}",
                                         base_ref.bold(),
                                         "push".bold(),
                                         push
                                     );
-                                        std::process::exit(2);
+                                            std::process::exit(2);
+                                        }
                                     }
                                 }
+                                _ => (),
                             }
-                            _ => {}
-                        },
+
+                            if !paths.is_empty() {
+                                let globset = {
+                                    let mut builder = GlobSetBuilder::new();
+                                    for path in paths.iter() {
+                                        builder.add(Glob::new(path)?);
+                                    }
+                                    builder.build()?
+                                };
+                                let changed_files = match (
+                                    std::env::var("CICADA_GIT_BASE"),
+                                    std::env::var("CICADA_GIT_HEAD"),
```

Comment:
These two new variables should address the problem. The `CICADA_GIT_BASE` should contain the hash of the latest commit on which cicada was run (for the same branch/pr), while `CICADA_GIT_HEAD` should be the hash of the commit that triggered the run.